File: Features/postgres-cordapp/workflows/src/test/java/net/corda/samples/postgres/flows/FlowTests.java
package net.corda.samples.postgres.flows;

import com.google.common.collect.ImmutableList;
import net.corda.core.concurrent.CordaFuture;
import net.corda.core.identity.CordaX500Name;
import net.corda.core.transactions.SignedTransaction;
import net.corda.testing.node.*;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

public class FlowTests {
    private final MockNetwork network = new MockNetwork(new MockNetworkParameters().withCordappsForAllNodes(ImmutableList.of(
            TestCordapp.findCordapp("net.corda.samples.postgres.contracts"),
            TestCordapp.findCordapp("net.corda.samples.postgres.flows")
    )).withNotarySpecs(ImmutableList.of(new MockNetworkNotarySpec(CordaX500Name.parse("O=Notary,L=London,C=GB")))));
    private final StartedMockNode a = network.createNode();
    private final StartedMockNode b = network.createNode();

    public FlowTests() {
        ImmutableList.of(a, b).forEach(it -> {
            it.registerInitiatedFlow(YoFlowResponder.class);
        });
    }

    @Before
    public void setup() {
        network.runNetwork();
    }

    @After
    public void tearDown() {
        network.stopNodes();
    }

    @Test
    public void emptyTest() { // ensures configuration is correct
    }

    //The logging flow should not have any input
    //This test will check if the input list is empty
    @Test
    public void dummyTest() throws ExecutionException, InterruptedException {
        CordaFuture<SignedTransaction> future = a.startFlow(new YoFlow(b.getInfo().getLegalIdentities().get(0)));
        network.runNetwork();
        SignedTransaction ptx = future.get();
        assert (ptx.getTx().getInputs().isEmpty());
    }
}

File: kotlin-samples/Features/dockerform-yocordapp/workflows/src/test/kotlin/net/corda/samples/dockerform/flows/FlowTests.kt
package net.corda.samples.dockerform.flows


import net.corda.core.identity.CordaX500Name
import net.corda.testing.node.*
import org.junit.After
import org.junit.Before
import org.junit.Test
import java.util.concurrent.ExecutionException


class FlowTests {

    lateinit var mockNetwork: MockNetwork
    lateinit var a: StartedMockNode
    lateinit var b: StartedMockNode

    @Before
    fun setup() {

        val mockNetwork = MockNetwork(MockNetworkParameters(cordappsForAllNodes = listOf(
                TestCordapp.findCordapp("net.corda.samples.dockerform.contracts"),
                TestCordapp.findCordapp("net.corda.samples.dockerform.flows")
        ),
                notarySpecs = listOf(MockNetworkNotarySpec(CordaX500Name("Notary","London","GB")))
        ))


        a = mockNetwork.createNode(MockNodeParameters())
        b = mockNetwork.createNode(MockNodeParameters())
        val startedNodes = arrayListOf(a, b)
        // For real nodes this happens automatically, but we have to manually register the flow for tests
        startedNodes.forEach { it.registerInitiatedFlow(YoFlowResponder::class.java) }
        mockNetwork.runNetwork()
    }

    @After
    fun tearDown() {

    }

    @Test
    fun `dummy test`() {
    }

    @Throws(ExecutionException::class, InterruptedException::class)
    fun dummyTest() {
        val future = a.startFlow(YoFlow(b.info.legalIdentities.first()))
        mockNetwork.runNetwork()
        val ptx = future.get()
        if (ptx != null) {
            assert(ptx.tx.inputs.isEmpty())
        }
    }
}
